packages/teleport/src/services/history/history.ts

```diff
@@ -114,16 +112,37 @@ const history = {
 const withBaseUrl = (url: string) => cfg.baseUrl + url;
 
 const match = (url: string) => (route: string) => {
-  return matchPath(url, {
-    path: route,
-    exact: true,
-  });
+  return matchPath(url, { path: route });
 };
 
 export default history;
 
-export function getUrlParameter(name = '', path = '') {
-  const params = new URLSearchParams(path);
-  const value = params.get(name);
-  return value || '';
+// regexFilterQueryParam is a regex to split a path into two groups.
+// First group captures anything before the filter param as the path we will search
+// for the target query param. The second group captures the first filter param and
+// anything else following it we assume is part of the filter.
+const regexFilterQueryParam = /(?<pathToSearch>.+?)(?<pathWithFilter>\?filter=.*)/;
+
+export function getUrlParameter(name = '', path = '', preserveFilter = false) {
```

Comment:
IMO we don't need to call this function without params so I'd remove default values for `name` and `path`